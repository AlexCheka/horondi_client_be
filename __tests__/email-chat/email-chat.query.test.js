/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const client = require('../../utils/apollo-test-client');
const { newQuestion, badQuestionId } = require('./email-chat.variables');

let questionId;

describe('Questions queries', () => {
  beforeAll(async () => {
    const createQuestion = await client.mutate({
      mutation: gql`
        mutation($question: EmailQuestionInput!) {
          addEmailQuestion(question: $question) {
            _id
          }
        }
      `,
      variables: {
        question: newQuestion,
      },
    });
    questionId = createQuestion.data.addEmailQuestion._id;
  });
  test('#1 Should receive all questions', async () => {
    const questions = await client.query({
      query: gql`
        query {
          getAllEmailQuestions {
            _id
            senderName
            text
            date
            email
            status
          }
        }
      `,
    });
    const allquestions = questions.data.getAllEmailQuestions;
    expect(allquestions).toBeDefined();
    expect(allquestions.length).toBeGreaterThan(0);
    expect(allquestions).toMatchSnapshot();
  });
  test('#2 Should receive question by ID', async () => {
    const createdQuestion = await client.query({
      query: gql`
        query($id: ID!) {
          getEmailQuestionById(id: $id) {
            ... on EmailQuestion {
              _id
              senderName
              text
              date
              email
              status
              answer {
                text
                date
              }
            }
            ... on Error {
              statusCode
              message
            }
          }
        }
      `,
      variables: { id: questionId },
    });
    const resultQuestion = createdQuestion.data.getEmailQuestionById;
    expect(resultQuestion).toBeDefined();
    expect(resultQuestion).toHaveProperty('senderName', 'Peter the Rabbit');
    expect(resultQuestion).toHaveProperty('text', 'I like your aplication!!!');
    expect(resultQuestion.date).toBeDefined();
    expect(resultQuestion).toHaveProperty('email', 'vovasays@goodbye.com');
    expect(resultQuestion).toHaveProperty('status', 'PENDING');
    expect(resultQuestion.answer).toBeNull();
  });
  test('#3 Should receive error if product ID is wrong', async () => {
    const badQuestion = await client.query({
      query: gql`
        query($id: ID!) {
          getEmailQuestionById(id: $id) {
            ... on EmailQuestion {
              _id
              senderName
              text
              date
              email
              status
              answer {
                text
                date
              }
            }
            ... on Error {
              statusCode
              message
            }
          }
        }
      `,
      variables: { id: badQuestionId },
    });
    const receivedError = badQuestion.data.getEmailQuestionById;
    expect(receivedError).toBeDefined();
    expect(receivedError).toHaveProperty('statusCode', 404);
    expect(receivedError).toHaveProperty('message', 'QUESTION_NOT_FOUND');
  });
  afterAll(async () => {
    await client.mutate({
      mutation: gql`
        mutation($id: ID!) {
          deleteEmailQuestion(id: $id) {
            ... on EmailQuestion {
              _id
            }
            ... on Error {
              statusCode
              message
            }
          }
        }
      `,
      variables: { id: questionId },
    });
  });
});
