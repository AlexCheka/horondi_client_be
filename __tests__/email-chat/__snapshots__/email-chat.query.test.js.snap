// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions queries #1 Should receive all questions 1`] = `
Array [
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f592a850acb7820e81eb3ca",
    "date": "1599679109473",
    "email": "hello@123132132",
    "senderName": "world",
    "status": "PENDING",
    "text": "greate expiriance",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f592b3d2efb2e44fcfb5d92",
    "date": "1599679293415",
    "email": "hello@123132132",
    "senderName": "Name is...",
    "status": "ANSWERED",
    "text": "greate expiriance",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f59feb47969d94bd85f60df",
    "date": "1599733428926",
    "email": "helloFBI@fbi.com",
    "senderName": "FBI is looking for you",
    "status": "PENDING",
    "text": "You can run but you can\`t hide",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f59fed97969d94bd85f60e0",
    "date": "1599733465806",
    "email": "helloFBI@fbi.com",
    "senderName": "FBI is looking for you",
    "status": "PENDING",
    "text": "You can run but you can\`t hide",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f5a026d0a675e4228bebbcd",
    "date": "1599734381251",
    "email": "helloFBI@fbi.com",
    "senderName": "FBI is looking for you",
    "status": "PENDING",
    "text": "You can run but you can\`t hide. Ha-ha",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f5a0370ad4e023f5498f462",
    "date": "1599734640899",
    "email": "helloFBI@fbi.com",
    "senderName": "FBI is looking for you",
    "status": "PENDING",
    "text": "You can run but you can\`t hide. Ha-ha",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f5a038959b54a20bc221b29",
    "date": "1599734665358",
    "email": "helloFBI@fbi.com",
    "senderName": "FBI is looking for you",
    "status": "SPAM",
    "text": "You can run but you can\`t hide. Ha-ha",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f60dd430716020830bd9a31",
    "date": "1600183619689",
    "email": "sgsgs@jshd.com",
    "senderName": "John",
    "status": "PENDING",
    "text": "hello",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f6120f71020142d6080343d",
    "date": "1600200951098",
    "email": "sgsgs@jshd.com",
    "senderName": "Peter",
    "status": "PENDING",
    "text": "helloWorld",
  },
  Object {
    "__typename": "EmailQuestion",
    "_id": "5f6133c8e1ab4d128cf93ad9",
    "date": "1600205768099",
    "email": "vovasays@goodbye.com",
    "senderName": "Peter the Rabbit",
    "status": "PENDING",
    "text": "I like your aplication!!!",
  },
]
`;
